### PR TITLE
Remove need of proguard config for DCE pass

### PR DESCRIPTION
**Pull request by parind**
1. Add -pthread to compile without errors for cmake.
2. Remove proguard config check from LocalDCEPass. Thanks @justinjhendrick.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,16 +8,22 @@ if (NOT CMAKE_BUILD_TYPE)
...
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
 if(BUILD_TYPE STREQUAL Static)
     set(ENABLE_STATIC ON CACHE BOOL "" FORCE)
 elseif (BUILD_TYPE STREQUAL Shared)
     set(ENABLE_STATIC OFF CACHE BOOL "" FORCE)
 endif ()
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
```
Comment (justinjhendrick) [inline CMakeLists.txt]: Just curious, why do we need to set this three times?

And would this fit better inside `set_common_cxx_flags_for_redex`?

**Comment by parind** [comment]
I am not sure if it would fit better inside "set_common_cxx_flags_for_redex". I guess one "SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")" could be enough.

File: CMakeLists.txt (hunk 1 above)
Comment (parind) [reply CMakeLists.txt]: I am still pretty new in this stuff, but I was wondering if we could use "set_common_cxx_flags_for_redex -pthread". Would this work ? I could try.

Comment (parind) [reply CMakeLists.txt]: @justinjhendrick I have made a new pull request with all the updates as of on this day.

**Comment by justinjhendrick** [comment]
Replaced by https://github.com/facebook/redex/pull/427/

**Pull request closed without merging**